beacon-chain/blockchain/receive_attestation.go

```diff
@@ -161,9 +161,25 @@ func (s *Service) spawnProcessAttestationsRoutine(stateFeed *event.Feed) {
 					log.WithError(err).Errorf("Unable to get justified balances for root %v", justified.Root)
 					continue
 				}
+				prevHead := s.headRoot()
 				if err := s.updateHead(s.ctx, balances); err != nil {
 					log.WithError(err).Warn("Resolving fork due to new attestation")
 				}
+				if s.headRoot() != prevHead {
+					finalized := s.store.FinalizedCheckpt()
+					if finalized == nil {
+						log.WithError(errNilFinalizedInStore).Error("Could not get finalized checkpoint")
+						continue
+					}
+					_, err := s.notifyForkchoiceUpdate(s.ctx,
+						s.headBlock().Block(),
+						s.headRoot(),
+						bytesutil.ToBytes32(finalized.Root),
+					)
+					if err != nil {
+						log.WithError(err).Error("could not notify forkchoice update")
+					}
+				}
```

Comment:
I recommend that we add a unit test to this, and It might make your life easier if you refactor this into its own method :)